api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/GraphTrainableModel.kt

```diff
@@ -900,4 +902,26 @@ public abstract class GraphTrainableModel(vararg layers: Layer) : TrainableModel
     override fun toString(): String {
         return "GraphTrainableModel(numberOfLayers=${layers.size}) ${super.toString()}"
     }
+
+    public override fun summary(): ModelSummary {
+        check(isModelCompiled) { "The model is not compiled yet. Compile the model to use this method." }
+
+        val (trainableLayers, frozenLayers) = layers.partition { it.isTrainable }
+
+        return ModelSummary(
+            type = this::class.simpleName.toString(),
+            name = name,
+            layersSummaries = layers.map { layer ->
+                LayerSummary(
+                    name = layer.name,
+                    type = layer::class.simpleName.toString(),
+                    outputShape = layer.outputShape,
+                    paramsCount = layer.paramCount.toLong(),
+                    inboundLayers = layer.inboundLayers.map { it.name }
+                )
+            },
+            trainableParamsCount = trainableLayers.sumOf { it.paramCount.toLong() },
+            frozenParamsCount = frozenLayers.sumOf { it.paramCount.toLong() },
```

Comment:
looks like paramCount is wrong in both cases: let me explain, all params from frozen layers are frozenParams and all params from not frozen, trainable layers are trainable params, revisit the calculation in the removed code